includes/class-wsuwp-shortcode-fundselector.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * plugin class file for the fundselector shortcode
+ *
+ * Defines the functions necessary to register our custom post types with
+ * WordPress.
+ *
+ * @link       https://github.com/washingtonstateuniversity/WSUWP-Plugin-iDonate
+ * @since      1.0.0
+ *
+ * @package    WSUWP_Plugin_iDonate_ShortCode_Fund_Selector
+ * @author     Blair Lierman
+ */
+class WSUWP_Plugin_iDonate_ShortCode_Fund_Selector {
+
+	/**
+	 * Initializes the plugin by registering the hooks necessary
+	 * for creating our custom post type within WordPress.
+	 *
+	 * @since    1.0.0
+	 */
+	public function init() {
+
+		add_shortcode( 'idonate_fundselector', array( $this, 'fundselector_create_shortcode' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'wsuf_fundselector_enqueue_scripts' ) );
+
+	}
+
+	// [idonate_fundselector embed="iDonate-embed-guid"]
+	public function fundselector_create_shortcode( $atts ) {
+		// $args = shortcode_atts( array(
+		// 	'embed' => 'something',
+		// ), $atts );
+
+		$return_string = '<div id="fundSelectionForm">';
+
+		// Major Categories button group
+		$return_string .= '
+		<div id="majorcategory" class="" role="group" aria-label="Category Selection Group">
+			<a class="active" role="button" href="#" data-tab="prioritiesTab">Priorities</a>
+			<a class="" role="button" data-category="programs" href="#">Programs</a>
+			<a class="" role="button" data-category="colleges" href="#">Colleges</a>
+			<a class="" role="button" data-category="campuses" href="#">Campuses</a>
+		</div>';
+
+		// Priorities Tab
+		$priorities = $this->wsuf_fundselector_funds_get_funds( 'idonate_priorities', 'idonate_priorities' );
+		$priorities_list = '<option disabled selected value> -- Select a Fund -- </option>';
+
+		foreach ( $priorities as $priority ) {
+			$fund_name = $priority['fund_name'];
+			$fund_designation_id = $priority['designation_id'];
+			$priorities_list .= "<option value=\"{$fund_designation_id}\">{$fund_name}</option>";
```

Comment:
Use `esc_attr()` when outputting data like `$fund_designation_id` into an HTML attribute. `esc_html()` should work (unless you're allowing HTML tags) for `$fund_name`. In both cases, the data should be escaped before output.